src/python/tests/assembly/test_download.py

```diff
@@ -0,0 +1,458 @@
+@pytest.mark.dependency(name="test_get_files_selection")
+@pytest.mark.parametrize(
+    "has_download_dir, files_expected, expectation",
+    [
+        pytest.param(
+            True,
+            {
+                "report": "GCA_017607445.1_ASM1760744v1_assembly_report.txt",
+                "fasta_dna": "GCA_017607445.1_ASM1760744v1_genomic.fna.gz",
+                "fasta_pep": "GCA_017607445.1_ASM1760744v1_protein.faa.gz",
+                "gff3_raw": "GCA_017607445.1_ASM1760744v1_genomic.gff.gz",
+                "gbff": "GCA_017607445.1_ASM1760744v1_genomic.gbff.gz",
+            },
+            does_not_raise(),
+            id="Normal case, data dir provided",
+        ),
+        pytest.param(
+            False,
+            {},
+            pytest.raises(FileDownloadError),
+            id="Error case, data dir not provided",
+        ),
+    ],
+)
+def test_get_files_selection(
+    data_dir: Path, has_download_dir: bool, files_expected: dict, expectation: ContextManager
+) -> None:
+    """Tests the `download.get_files_selection()` function.
+
+    Args:
+        download_dir: Path to specific location of downloaded files.
+        files_expected: Defines contents of test files downloaded
+        expectation: Context manager expected raise exception
+    """
+
+    if has_download_dir:
+        download_dir = data_dir
+    else:
+        download_dir = Path()
+
+    with expectation:
+        subset_files = get_files_selection(download_dir)
+        for file_end_name, file_path in subset_files.items():
+            expected_file = files_expected[file_end_name]
+            test_data_file_name = Path(file_path).name
+            assert test_data_file_name == expected_file
+
+
+##################
+@pytest.mark.dependency(name="test_retrieve_assembly_data")
+@pytest.mark.parametrize(
+    "accession, is_dir, files_downloaded, md5_return, exception",
+    [
+        pytest.param(
+            "GCA_017607445.1",
+            True,
+            {
+                "report": "GCA_017607445.1_ASM1760744v1_assembly_report.txt",
+                "fasta_dna": "GCA_017607445.1_ASM1760744v1_genomic.fna.gz",
+                "fasta_pep": "GCA_017607445.1_ASM1760744v1_protein.faa.gz",
+                "gff3_raw": "GCA_017607445.1_ASM1760744v1_genomic.gff.gz",
+                "gbff": "GCA_017607445.1_ASM1760744v1_genomic.gbff.gz",
+            },
+            True,
+            does_not_raise(),
+            id="Case 1: Good accession, dir exists",
+        ),
+        pytest.param(
+            "GCA_017607445.1",
+            False,
+            0,
+            False,
+            pytest.raises(FileExistsError),
+            id="Case 2: Good accession, Not a dir",
+        ),
+        pytest.param(
+            "GCA_017607445.1",
+            True,
+            0,
+            False,
+            pytest.raises(FileDownloadError),
+            id="Case 3: Download files list empty",
+        ),
+    ],
+)
+@patch("ftplib.FTP")
```

Comment:
You need to patch the FTP object that you use in the module (autospec ensures all its methods are also mocked):
```suggestion
@patch("ensembl.io.genomio.assembly.download.FTP", autospec=True)
```